trin-core/src/utp/mod.rs

```diff
@@ -0,0 +1,3 @@
+#[allow(clippy::module_inception)]
+pub mod utp;
```

Comment:
Certainly, we want to rethink any choices that cause clippy warnings. `utp` in utp does seem like a surprising naming choice. Can we do better so that we don't need this clippy silence?

Reply:
Ok I refactored utp so this is no longer an issue. I want to rework the utp implementation so it doesn't rely on and async as much where it doesn't need to.